mmdet/core/bbox/transforms.py

```diff
@@ -47,14 +73,19 @@ def delta2bbox(rois,
     max_ratio = np.abs(np.log(wh_ratio_clip))
     dw = dw.clamp(min=-max_ratio, max=max_ratio)
     dh = dh.clamp(min=-max_ratio, max=max_ratio)
+    # Compute center of each anchor ROI grid cell
```

Comment:
Makes sense. I think I was a bit confused on the details of this function before, but I'm starting to understand it better now. 

I'm still unsure of two things: First, In what circumstances would you want nonzero means and non-unity stds? What is the denormalization for?

Second, the logic for shifting the center of the boxes seems to allow the boxes to move anywhere in the image. 
 
```python
    gx = torch.addcmul(px, 1, pw, dx)  # gx = px + pw * dx
    gy = torch.addcmul(py, 1, ph, dy)  # gy = py + ph * dy
```

I'm most familiar with the YOLO detector, which bounds how far the center of an anchor box can move using logic similar to this (where W and H are the maximum distances an anchor can move): 

 ```python
    gx  = px + W * (dx.sigmoid() - 0.5)
    gy = py + H * (dy.sigmoid() - 0.5)
```

Can you elaborate on the reason for using the first method over the second?

Reply:
1. YOLO is more different from other detectors. In Faster R-CNN or SSD, nonzero means and non-unity stds are used for regression by default. The regression target is normalized with precomputed means and stds, so that we need to denormalize the network predictions.
2. The logic also follows the original papers. In this step the shift can be arbitrary, and may be clipped to the image shape later on.

Reply:
This makes sense. I added in a reference to https://arxiv.org/abs/1311.2524, which I is the paper FasterRCNN cites about using this delta encoding. That should help anyone else who reads the docs and has a similar question.